sympy/solvers/tests/test_diophantine.py

```diff
@@ -273,11 +273,9 @@ def test_bf_pell():
 def test_length():
     assert length(2, 1, 0) == 1
```

Comment:
These tests are more than sufficient to cover all cases for the length function which must detect when there is or isn't a repeat part:
```
(2, 1, 0)   [2]
(-2, 4, 5)  [[16, 1]]
(-5, 4, 17) [0, [3, 1, 1]]
(0, 4, 13)  [1, [9, 7, 9, 2]]
(7, 13, 11) [1, 3, [1, 5, 2, 3, 2, 5, 1, 2, 1, 1, 1, 1, 8, 86, 8, 1, 1, 1, 1, 2]]
(1, 6, 4)   [0, 2]
```